subsys/bluetooth/host/hci_core.c

```diff
@@ -3627,6 +3629,49 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+int bt_disable(void)
+{
+	int err;
+
+	if (!bt_dev.drv) {
+		BT_ERR("No HCI driver registered");
+		return -ENODEV;
+	}
+
+	if (!bt_dev.drv->close) {
+		return -ENOTSUP;
+	}
+
+	if (atomic_test_and_set_bit(bt_dev.flags, BT_DEV_DISABLE)) {
+		return -EALREADY;
+	}
+
+	err = bt_dev.drv->close();
+	if (err) {
+		BT_ERR("HCI driver close failed (%d)", err);
+		return err;
+	}
+
+	/* Abort TX thread */
+	k_thread_abort(&tx_thread_data);
+
+#if !defined(CONFIG_BT_RECV_IS_RX_THREAD)
+	/* Abort RX thread */
+	k_thread_abort(&rx_thread_data);
+#endif
+
+	if (IS_ENABLED(CONFIG_BT_TINYCRYPT_ECC)) {
+		bt_hci_ecc_deinit();
+	}
+
+	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
+
+	atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);
```

Comment:
I just took a look at #41424 and it's adding an API For checking BT_DEV_READY instead of BT_DEV_ENABLE. How do these relate? It seems that either this or that PR should be changed (ping @JordanYates), since if we don't clear BT_DEV_READY then the new `bool bt_dev_ready(void);` API that @JordanYates adds wouldn't work with `bt_disable()` as added here.

Reply:
Clearing `BT_DEV_READY` should definitely be done. Most Bluetooth API's are checking whether `BT_DEV_READY` is set, not `BT_DEV_ENABLE`.

Reply:
@jhedberg I've added BT_DEV_READY clearing.

Reply:
@jhedberg I've fixed the point.